File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/UserReplaceCommand.cs
using SIE.Andon.Andons;
using SIE.Andon.Andons.ViewModels;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.Andon.Andons.Commands
{
    /// <summary>
    /// 责任人替换
    /// </summary>
    public class UserReplaceCommand : ViewCommand<UserReplaceViewModel>
    {
        protected override object Excute(UserReplaceViewModel args, string scope)
        {
            RT.Service.Resolve<AndonController>().ReplaceDetailUser(args);
            return true;
        }
    }
}

File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/UserReplaceCommand.js
SIE.defineCommand('SIE.Web.Andon.Andons.Commands.UserReplaceCommand', {
    meta: { text: "责任人替换", group: "edit", iconCls: "icon-Handshake icon-blue" },
    canExecute: function (view) {
        return true;
    },

    execute: function (view) {
        var me = this;
        SIE.AutoUI.getMeta({
            model: 'SIE.Andon.Andons.ViewModels.UserReplaceViewModel',
            isDetail: true,
            ignoreQuery: true,
            async: false,
            module: 'SIE.Andon.Andons.AndonGroup,SIE.Andon',
            token: this._token,
            callback: function (res) {
                var mainBlock;
                if (res.mainBlock)
                    mainBlock = res.mainBlock;
                else
                    mainBlock = res;
                var detailView = SIE.AutoUI.createDetailView(mainBlock);
                var entity = new detailView._model();
                detailView._setDefaultValue(entity);
                detailView.setData(entity);
                var ui = detailView.getControl();

                var win = SIE.Window.show({
                    title: '责任人替换'.t(),
                    width: 600,
                    height: 200,
                    items: ui,
                    buttons: ['确认', '取消'],
                    callback: function (btn) {
                        if (btn == '确认') {
                            me.view.execute({
                                data: entity.data,
                                success: function (rst) {
                                    me.view.reloadData();
                                }
                            });
                        }
                    }
                });
            }
        });
    }
});
